engine/class_modules/priest/sc_priest.cpp

```diff
@@ -144,6 +175,12 @@ struct mind_blast_t final : public priest_spell_t
       }
 
       priest().buffs.coalescing_shadows->expire();
+      if ( priest().talents.discipline.expiation.enabled() )
+      {
+        impact_action = new expiation_t( priest() );
```

Comment:
instead of creating the spell here I would prefer adding this to the constructor, should be a bit faster that way as well. We do this for things like `vampiric_touch_t` if you want to see a few examples.